post-processing/plot_handler.py

```diff
@@ -12,6 +13,29 @@
 from titlecase import titlecase
 import itertools
 
+def get_axis_min_max(df, axis, column):
+    axis_range = axis["range"]
+    axis_min = axis_range["min"] if axis_range["min"] != 'None' else None
+    axis_max = axis_range["max"] if axis_range["max"] != 'None' else None
+        
+    #FIXME: str types and user defined datetime ranges not currently supported
+    # use defaults if type is datetime
+    axis_min_element = np.nanmin(df[column])
+    axis_max_element = np.nanmax(df[column])
+    if (is_datetime(df[column])):
+        datetime_range = axis_max_element - axis_min_element
+        buffer_time = datetime_range*0.2
+        axis_min = axis_min_element - buffer_time
+        axis_max = axis_max_element + buffer_time
+    else:
+        if not (axis_min and axis_max):
+            axis_min = (axis_min_element*0.6 if min(df[column]) >= 0
+                    else math.floor(axis_min_element*1.2))
+            axis_max = (axis_max_element*0.6 if max(df[column]) <= 0
+                    else math.ceil(axis_max_element*1.2))
```

Comment:
Good idea, but having tried this in practice custom ranges don't work if you use 0 as one of your values (which is a valid value to want for either your min or max), so I think you might need a more specific check for this if statement.

We also get some interesting behaviour if you set your min value to be greater than your max value: the plot gets mirrored, which isn't necessarily a bad thing as long as it's not breaking anything, but definitely worth making a note of this somewhere. A tooltip maybe? Not sure if Streamlit allows you to attach tooltips to checkboxes though...

Reply:
Ah that is annoying about zero not being allowed. but you can have (+-)0.01 which could possibly cover the needed functionality, although i agree it's not ideal and a tooltip would definitely be needed.

I suppose the mirroring axes is just the same as the sort descending functionality ,so it might not be so bad?

Reply:
I tried to do this but couldn't get it working do I'll add a FIXME comment